framework/src/headers/mac.rs

```diff
@@ -153,3 +166,61 @@ impl MacHeader {
         self.dst.copy_address(&src);
     }
 }
+
+#[derive(Debug, PartialEq)]
+pub struct MacParseError(());
+
+impl fmt::Display for MacParseError {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        fmt.write_str(self.description())
+    }
+}
+
+impl Error for MacParseError {
```

Comment:
do we want to place the error as part of the generalized error_chain, instead of the separate struct?

Reply:
yes, yes, definitely. had forgotten that was being used in netbricks too. :+1:

Reply:
actually, not so sure. `FromStr` requires an associated type that implements `'static Sized`, which I'm not sure we can get out of the error chain stuff. Might need to just use this `MacParseError` and foreign_link it elsewhere. :| Will play more with it after lunch.

Reply:
ah, hmm. good catch @clinedome. Yeah, take a gander just to be sure (via the lib). No biggie.